### PR TITLE
[DEPRECATED] Adds natspec documentation to the vouching package

### LINKED ISSUE #147: Add contracts inline documentation
Add a section to output contracts inline documentation. 
We should add documentation for both lib contracts and token/vouching contracts.

### PR DESCRIPTION
**Pull request by ajsantander**
DO NOT MERGE
A cleaner version of this PR is https://github.com/zeppelinos/zos/pull/347

Fixes #147 

Also publishes the content on the docs site. 

@elopio could you please review the content of the natspec?
@jcarpanelli could you please review the publication of the natspec in the docsite? `genVouchDocs()` in `zos/packages/docs/scripts/bump-docs.js`

### CODE REVIEW

File: packages/docs/scripts/bump-docs.js [hunk 1]
```diff
@@ -65,6 +66,27 @@ function genCliDocs (packagesDir) {
   return { 'CLI REFERENCE': commands }
 }
 
+function genVouchDocs(packagesDir) {
+  const docsDir = path.resolve(packagesDir, 'docs', 'docs')
+  const builtDocs = path.resolve(docsDir, 'docs')
+  const vouchDir = path.resolve(packagesDir, 'vouching')
+  const vouchContractsDir = path.resolve(vouchDir, 'contracts')
+  const libDir = path.resolve(vouchDir, 'node_modules', 'zos-lib')
+  const tplDir = path.resolve(vouchDir, 'node_modules', 'tpl-contracts-zos')
```
Comment (jcarpanelli) [inline packages/docs/scripts/bump-docs.js]: Shouldn't we change this to `tpl-contracts-eth`?
```suggestion
  const tplDir = path.resolve(vouchDir, 'node_modules', 'tpl-contracts-eth')
```

File: packages/docs/scripts/bump-docs.js [hunk 2]
```diff
@@ -65,6 +66,27 @@ function genCliDocs (packagesDir) {
...
+  const ozeDir = path.resolve(vouchDir, 'node_modules', 'openzeppelin-zos')
```
Comment (jcarpanelli) [inline packages/docs/scripts/bump-docs.js]: Same here
```suggestion
  const ozeDir = path.resolve(vouchDir, 'node_modules', 'openzeppelin-eth')
```

File: packages/docs/scripts/bump-docs.js [hunk 3]
```diff
@@ -65,6 +66,27 @@ function genCliDocs (packagesDir) {
...
+  const sidebar =  path.resolve(docsDir, 'website', 'sidebars.json')
+
+  cd(vouchDir)
+  exec('npm install > "/dev/null" 2>&1')
+  exec(`SOLC_ARGS='openzeppelin-zos=${ozeDir},zos-lib=${libDir},tpl-contracts-zos=${tplDir}' npx solidity-docgen ${vouchDir} ${vouchContractsDir} ${docsDir} --exclude mocks`)
```
Comment (jcarpanelli) [inline packages/docs/scripts/bump-docs.js]: Are we sure that `solidity-docgen` supports multiple external libraries?

File: packages/vouching/contracts/Vouching.sol [hunk 4]
```diff
@@ -7,53 +7,122 @@ import "openzeppelin-zos/contracts/token/ERC20/SafeERC20.sol";
 import "openzeppelin-zos/contracts/math/SafeMath.sol";
 import "openzeppelin-zos/contracts/utils/Address.sol";
 
+/**
+ * @title Vouching
+ * @dev Contract for staking tokens against dependencies.
+ * TODO
```
[the remark this answers is not in the document]
Comment (ajsantander) [reply packages/vouching/contracts/Vouching.sol]: I'm not sure I put that there, but removing it.

File: packages/vouching/contracts/Vouching.sol [hunk 5]
```diff
@@ -7,53 +7,122 @@ import "openzeppelin-zos/contracts/token/ERC20/SafeERC20.sol";
...
+   */
   event DependencyCreated(
     bytes32 indexed nameHash,
     string name,
     address indexed owner,
     address indexed dependencyAddress,
     uint256 initialStake
   );
+  /**
+   * @dev Emitted when the owner of a dependency adds stake to it.
```
[the remark this answers is not in the document]
Comment (ajsantander) [reply packages/vouching/contracts/Vouching.sol]: Changed.

I tried to find other places where "add stake to" could be replaced with "vouch for" and I didn't find any. Furthermore, I tried to replaced "remove stake from" with some word representing the opposite of "vouch", but things like "abstain" or "disclaim" sound weird, so I didn't change that either.

TL;DR: Just changed this one case you mention here.

File: packages/vouching/contracts/Vouching.sol [hunk 6]
```diff
@@ -7,53 +7,122 @@ import "openzeppelin-zos/contracts/token/ERC20/SafeERC20.sol";
...
+   * @param nameHash bytes32 hash of the name of the dependency.
+   * @param amount uint256 with the amount of stake that has been added to the dependency.
+   */
   event Vouched(bytes32 indexed nameHash, uint256 amount);
+  /**
+   * @dev Emitted when the owner of a dependency removes stake from it.
```
[the remark this answers is not in the document]
Comment (ajsantander) [reply packages/vouching/contracts/Vouching.sol]: Not sure why this sounds super weird to me... maybe because owner is singular, their is plural?

File: packages/docs/scripts/bump-docs.js (hunk 1 above)
Comment (ajsantander) [reply packages/docs/scripts/bump-docs.js]: This should go away upon rebasing.

File: packages/docs/scripts/bump-docs.js (hunk 3 above)
Comment (ajsantander) [reply packages/docs/scripts/bump-docs.js]: I think so, but I'm pretty sure they have to be space separated instead of comma separated. So thx!

**Comment by facuspagnuolo** [comment]
Closing in favour https://github.com/zeppelinos/zos/pull/347
